docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,275 @@
+  quality, redirecting the debugging and troubleshooting efforts away from the
+  well tested libraries.
+
+The new standalone repository will have predefined pipelines to run those
+checks, so new libraries will be automatically enforced to newly introduced
+libraries.
+
+Metadata description
+--------------------
+
+Every module will need to provide a description including:
+
+* Utility description
+* The utility category (ie: network, cpu, security, etc..)
+* Maintainers information
+  * Main contact (We need a contact point, a single maintainer or a group)
+  * Maintainer Group members (this will be used to block merges)
+* Supported Platforms
+* Supported execution method (Local vs Remote)
+
+For instance, the textual category and description will allow users to search
+for modules based on their interest.  Also, our pipelines will parse that data
+field containing the list of supported platforms, and will trigger tests on
+those platforms. Users will then be able to see the results of those tests on
+each of the platforms defined, giving them quality assurance and coverage
+there.
+
+The exact format of the metadata file is still to be defined, but a possibility
+is to define a JSON schema that will be provided on this new repository.
+
+Maintainability
+===============
+
+As described above, each library must follow the new standard and keep that
+same level for future versions. Because of that, new PRs must be approved by at
+least one MAINTAINER and pass the basic checks pipelines that will be provided
+as part of the autils repository.
+
+Besides the automatic checks, each MAINTAINER is responsible for keeping the
+consistency of that library with the overall project.
+
+Release
+=======
+
+As a new and standalone repository, there will not be any type of release
+version sync between Avocado and autils.  Avocado may release, say, version
+100.0, while autils won’t have a release. Because of that, autils will be
+versioned at a different pace than Avocado.
+
+Every week a automatic job will be executed to determine if there is a
+potential need for a new release:
+
+* If the number of commits since last release has reached a threshold; AND
+* There is no open discussion with the tag “release-decision”:
+  * Creates a new discussion (with the "release-decision" tag) to run a poll
+    between all the MAINTAINERS, so they will have the opportunity to thumbs
+    up if there is a need for a new release;
+  * The link to the new discussion should be sent to all MAINTAINERS;
+  * If the decision is positive, a round-robin selected maintainer should be responsible for coordinating the release it self;
+  * And the discussion should be closed
+* If there is any existing open discussion, the bot job should comment on the
+  same discussion with the updated list of commits and ping again the
+  MAINTAINERS, give them the opportunity to thumbs up based on the new
+  status;
```

Comment:
Are there any other criteria for a release than just a threshold number of commits that was reached? E.g. security bug detected by GH or other bot or similar triggers?

Reply:
That would be a good indicator as well. Probably it will make no sense to wait for more commits if there is security bug fixed. The problem I see, is the automation here, since we don't have any errata mechanism in place (and probably don't need at this point).

So yes, we should accommodate those "manual" needs, as well. Thanks.